frontend/src/components/Login.jsx

```diff
@@ -1,89 +1,255 @@
+        method: "POST",
+        headers: {
+          "Content-Type": "application/x-www-form-urlencoded",
+          accept: "application/json",
+        },
+        body: requestBody,
+      });
+
+      if (!response.ok) {
+        setLoginError(true);
+        setIdentifierValidated(true);
+        setPasswordValidated(true);
+        const data = await response.json();
+        if (data && data.detail) {
+          setLoginErrorMessage(data.detail);
+        } else {
+          setLoginErrorMessage("Something went wrong. Please try again later.");
         }
-    };
-
-    return (
-        <div className="login-container">
-            <form onSubmit={handleSubmit}>
-                <img src={logo} alt="Logo" className="logo_login" />
-                
-                <div className="password-container">
-                      <input 
-                          type="text" 
-                          placeholder="Usuari" 
-                          value={username} 
-                          onChange={e => setUsername(e.target.value)}
-                      />
-                </div>
-
-                <div className="password-container">
-                    <input 
-                        type={showPassword ? "text" : "password"} 
-                        placeholder="Contrasenya" 
-                        value={password} 
-                        onChange={e => setPassword(e.target.value)}
+      } else {
+        const data = await response.json();
+        setToken(data.access_token);
+        navigate("/userfeed");
+      }
+    } catch (error) {
+      setLoginError(true);
+      setLoginErrorMessage("Something went wrong. Please try again later.");
+    }
+  };
+
+  /* Events */
+
+  const onIdentifierChange = (e) => {
+    setIdentifier(e.target.value);
+    setIdentifierFilled(e.target.value.length > 0);
+    setIdentifierValidated(false);
+  };
+
+  const onPasswordChange = (e) => {
+    setPassword(e.target.value);
+    setPasswordFilled(e.target.value.length > 0);
+    setPasswordValidated(false);
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    if (identifier.length == 0) {
+      setIdentifierFilled(false);
+      setIdentifierValidated(true);
+    }
+    if (password.length == 0) {
+      setPasswordFilled(false);
+      setPasswordValidated(true);
+    }
+
+    if (identifierFilled && passwordFilled) {
+      try {
+        loginRequest(identifier, password);
+      } catch (error) {
+        console.log(error);
+      }
+    }
+  };
+
+  return (
+    <div>
+      <Container className="py-5 min-vh-100">
+        <Row>
+          <Col sm={4}></Col>
+          <Col sm={4} className="form-container rounded">
+            <Row>
+              <Col sm={1}>
+                <Button
+                  variant="link"
+                  className="text-decoration-none fs-3 text-reset my-2"
+                  onClick={() => navigate("/userfeed")}
```

Comment:
This appears to be a navigation button. Using the onClick handler to manage routing is straightforward, but it may be beneficial to use a Link component (if available) for semantic reasons.

Reply:
This component is incomplete, i left it as a button so it has the on hover style change, but since the color is black for now it is not apparent, we could change this in the future.